instrumentation/grape/lib/opentelemetry/instrumentation/grape/event_handler.rb

```diff
@@ -0,0 +1,133 @@
+            type = payload[:type]
+
+            # Prevent submitting empty filters
+            return if (!filters || filters.empty?) || !type || (finish - start).zero?
+
+            name = span_name(payload[:endpoint])
+            attributes = {
+              'component' => 'web',
+              'operation' => 'endpoint_run_filters',
+              'grape.filter.type' => type.to_s
+            }
+            tracer.in_span(name, attributes: attributes, start_timestamp: start, kind: :server) do |span|
+              handle_payload_exception(span, payload[:exception_object]) if payload[:exception_object]
+            end
+          end
+
+          # Handles the format_response.grape event
+          def format_response(_name, start, _finish, _id, payload)
+            endpoint = payload[:env]['api.endpoint']
+            name = span_name(endpoint)
+            attributes = {
+              'component' => 'template',
+              'operation' => 'format_response',
+              'grape.formatter.type' => formatter_type(payload[:formatter])
+            }
+            tracer.in_span(name, attributes: attributes, start_timestamp: start, kind: :server) do |span|
+              handle_payload_exception(span, payload[:exception_object]) if payload[:exception_object]
+            end
+          end
+
+          private
+
+          def tracer
+            Grape::Instrumentation.instance.tracer
+          end
+
+          def span_name(endpoint)
+            "#{request_method(endpoint)} #{path(endpoint)}"
+          end
+
+          def run_attributes(payload)
+            endpoint = payload[:endpoint]
+            path = path(endpoint)
+            {
+              'component' => 'web',
+              'operation' => 'endpoint_run',
+              'grape.route.endpoint' => endpoint.options[:for]&.base.to_s,
+              'grape.route.path' => path,
+              'grape.route.method' => endpoint.options[:method].first,
+              OpenTelemetry::SemanticConventions::Trace::HTTP_METHOD => request_method(endpoint),
+              OpenTelemetry::SemanticConventions::Trace::HTTP_ROUTE => path
+            }
+          end
+
+          # ActiveSupport::Notifications will attach a `:exception_object` to the payload if there was
+          # an error raised during the execution of the &block associated to the Notification.
+          # This can be safely added to the span for tracing.
+          def handle_payload_exception(span, exception)
+            span.record_exception(exception)
+            span.status = OpenTelemetry::Trace::Status.error("Unhandled exception of type: #{exception.class}")
+          end
+
+          def request_method(endpoint)
+            endpoint.options[:method]&.first
+          end
+
+          def path(endpoint)
+            namespace = endpoint.routes.first.namespace
+            version = endpoint.routes.first.options[:version] || ''
+            prefix = endpoint.routes.first.options[:prefix]&.to_s || ''
```

Comment:
This aspect of our implementation differs from DDog: I decided to include the prefix and the version in the full path, because they are part of the path so I don't see why we shouldn't. If you can think of any objections, please let me know 🙏 😄

Reply:
To give some extra context, I decided to add the prefix and version to be able to differentiate cases such as the following:

```ruby
class V1API < Grape::API
  prefix :api
  version 'v1'
  get :hello do
    { message: 'Hello, world!' }
  end
end

class V2API < Grape::API
  prefix :api
  version 'v2'
  get :hello do
    { message: 'Hello, new world!' }
  end
end
```

If these are not added, the path will be `/hello` for both, and I thought it would be helpful to be able to differentiate them. At Catawiki we have several cases like this one where it would be useful for debugging to know the full path.